Khan/wonder-blocks
# Link broken layout fixes

## Summary:
- Move `verticalAlign` to only apply to links `withIcon`
- Add check for external vs relative link; only display external icon for external links with `target="_blank"`
- Update external links in SB
- Update unit tests to reflect external link changes

Issue: n/a

## Test plan:
`yarn jest`
Go to Default Link in SB.

For `verticalAlign` update:
- Inspect Default Link and confirm `verticalAlign` is not set to `bottom`.
- In control panel, add a `startIcon` or an `endIcon`.
- Inspect Link again and see that `verticalAlign: bottom` appear.
- You can also inspect the links in the Inline story and compare `verticalAlign` for links with and without the external icon.

For external icon update:
- In control panel, change `href` to an external path and set `target` to `_blank`.
- An external icon should only appear if both are true.
- Icon should not appear if `href` is relative OR if `target` is not set to `_blank`.
